javascript/javascript3/week2/main.js

```diff
@@ -0,0 +1,259 @@
+// --------------------------------------------------------------------------------   Movies exercise
+
+let allbadmov = [];
+let allbad2000 = [];
+let allbadtitle2000 = [];
+
+function getmovielist(data) {
+  // sessionStorage.clear();
+  let selectedvalue = document.getElementById("movlist").value;
+  // created an inputbox and a button for searching the gif file
+
+  if (selectedvalue == 0) {
+    alert("This can't be blank");
+    document.getElementById("movselected").innerHTML = `You should select some option to see results`;
+    document.getElementById("movresult").innerHTML = ``; 
+
+  } else if (selectedvalue == "allbad") {
+
+    console.log(selectedvalue);
+    data.forEach(element => {
+
+      if (element.rating < 5) {
+        allbadmov.push(element);
+      
+      }
+    });
+
+    document.getElementById("movselected").innerHTML = `you have selected option no 1`;
+    document.getElementById("movresult").innerHTML = showtabulardata(allbadmov);
+
+  } else if (selectedvalue == "allbad2000") {
+    // Created an array of bad movies since year 2000
+    console.log(selectedvalue);
+
+    data.forEach(element => {
+
+      if (element.rating < 5 && element.year < 2000) {
+
+        for (let i = 0; i < element.length; i++) {
+        
+          const element = array[i];

```

Comment:
You have `element` in the forEach, and then this one here, which is super confusing. Also , this element is not used anywhere. Not sure what this code is trying to do.

Reply:
Sorted it, and updated it with map and ternary operators 👍